pkg/connector/openai/v0/main.go

```diff
@@ -35,39 +39,93 @@ var (
 	openAIJSON []byte
 
 	once sync.Once
-	con  *connector
+	con  *Connector
 )
 
-type connector struct {
+// Connector executes queries against OpenAI.
+type Connector struct {
 	base.BaseConnector
+
+	// Global secrets.
+	globalAPIKey string
 }
 
 type execution struct {
 	base.BaseConnectorExecution
 }
 
-func Init(l *zap.Logger, u base.UsageHandler) *connector {
+// Init returns an initialized OpenAI connector.
+func Init(l *zap.Logger, u base.UsageHandler) *Connector {
 	once.Do(func() {
-		con = &connector{
+		con = &Connector{
 			BaseConnector: base.BaseConnector{
 				Logger:       l,
 				UsageHandler: u,
 			},
 		}
+
 		err := con.LoadConnectorDefinition(definitionJSON, tasksJSON, map[string][]byte{"openai.json": openAIJSON})
 		if err != nil {
 			panic(err)
 		}
 	})
+
 	return con
 }
 
-func (c *connector) CreateExecution(sysVars map[string]any, connection *structpb.Struct, task string) (*base.ExecutionWrapper, error) {
+// The connection parameter is defined with snake_case, but the
+// environment variable configuration loader replaces underscores by dots,
+// so we can't use the parameter key directly.
+func readFromSecrets(key string, s map[string]any) string {
+	sanitized := strings.ReplaceAll(key, "_", "")
+	if v, ok := s[sanitized].(string); ok {
+		return v
+	}
```

Comment:
❓ If we used `camelCase` in connection params this could be simplified. But it requires a migration so I didn't take that path. Still interesting to consider, since we've recently discussed unifying cases in definition files.

Reply:
Yes, I think we'll use kebab-case for all component fields soon. We can remove this workaround after we change all the fields to kebab-case.